src/lxc/confile_utils.c

```diff
@@ -532,10 +532,44 @@ int rand_complete_hwaddr(char *hwaddr)
 	return 0;
 }
 
+bool lxc_config_net_hwaddr(const char *line)
+{
+	char *copy, *p;
+
+	if (strncmp(line, "lxc.net", 7) != 0)
+		return false;
+	if (strncmp(line, "lxc.network.hwaddr", 18) == 0)
+		return true;
+
+	/* We have to dup the line, if line is something like
+	 * "lxc.net.[i].xxx = xxxxx ", we need to remove
+	 * '[i]' and compare its key with 'lxc.net.hwaddr'*/
+	copy = strdup(line);
+	if (!copy) {
+		SYSERROR("failed to allocate memory");
+		return false;
+	}
+	if (*(copy + 8) >= '0' && *(copy + 8) <= '9') {
+		p = strchr(copy + 8, '.');
+		if (!p) {
+			free(copy);
+			return false;
+		}
+		strcpy(copy + 8, p + 1);
+	}
+	if (strncmp(copy, "lxc.net.hwaddr", 14) == 0) {
```

Comment:
Isn't this supposed to handle bot `lxc.net.[i].hwaddr` and `lxc.network.{[i]}.hwaddr.*`? If so you should add:

```
if (strncmp(copy, "lxc.network.hwaddr", appropriate-length)  == 0) {
        /* same error handling */
}
```
You can probably move this into a single line by using `&&`.